internal/controller/pkg/revision/reconciler.go

```diff
@@ -858,3 +886,41 @@ func (r *Reconciler) deactivateRevision(ctx context.Context, pr v1.PackageRevisi
 
 	return nil
 }
+
+func (r *Reconciler) runtimeManifestBuilderOptions(ctx context.Context, pwr v1.PackageRevisionWithRuntime) ([]RuntimeManifestBuilderOption, error) {
+	var opts []RuntimeManifestBuilderOption
+
+	if r.features.Enabled(features.EnableBetaDeploymentRuntimeConfigs) {
+		rcRef := pwr.GetRuntimeConfigRef()
+		if rcRef == nil {
+			return nil, errors.New(errNoRuntimeConfig)
+		}
+
+		rc := &v1beta1.DeploymentRuntimeConfig{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: rcRef.Name}, rc); err != nil {
+			return nil, errors.Wrap(err, errGetRuntimeConfig)
+		}
+		opts = append(opts, RuntimeManifestBuilderWithRuntimeConfig(rc))
+	}
+
+	cc := &v1alpha1.ControllerConfig{}
+	if ccRef := pwr.GetControllerConfigRef(); ccRef != nil {
+		if err := r.client.Get(ctx, types.NamespacedName{Name: ccRef.Name}, cc); err != nil {
+			return nil, errors.Wrap(err, errGetControllerConfig)
+		}
+		opts = append(opts, RuntimeManifestBuilderWithControllerConfig(cc))
+	}
```

Comment:
Would it make sense to add a comment somewhere in here explaining the relationship between these two. i.e. Is it possible for both options to be appended? What happens if they are?